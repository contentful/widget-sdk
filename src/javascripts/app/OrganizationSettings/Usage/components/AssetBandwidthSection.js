import React from 'react';
import { css } from 'emotion';
import PropTypes from 'prop-types';
import { partialRight } from 'lodash';
import {
  TextLink,
  Note,
  Typography,
  Heading,
  Paragraph,
} from '@contentful/forma-36-react-components';
import * as tokens from '@contentful/forma-36-tokens';

import { shortenStorageUnit } from 'utils/NumberUtils';

const styles = {
  heading: css({
    color: '#536171',
    fontWeight: tokens.fontWeightNormal,
  }),
  usageNumber: css({
    color: '#253545',
    fontSize: tokens.fontSize3Xl,
    fontWeight: tokens.fontWeightMedium,
  }),
  limit: css({
    fontSize: tokens.fontSizeM,
  }),
  includedLimit: css({
    color: '#6A7889',
  }),
  overage: css({
    color: tokens.colorOrangeMid,
  }),
  learnMoreLink: css({
    marginLeft: tokens.spacingS,
  }),
  note: css({
    width: '40%',
    marginTop: tokens.spacingS,
    color: '#27457A',
  }),
};

const AssetBandwidthSection = ({ limit, usage, uom }) => {
  const withUnit = partialRight(shortenStorageUnit, uom);

  return (
    <Typography>
      <Heading element="h2" className={styles.heading}>
        Total asset bandwidth
      </Heading>
      <Paragraph data-test-id="asset-bandwidth-usage" className={styles.usageNumber}>
        {withUnit(usage)}
      </Paragraph>
      <div className={styles.limit}>
        <strong data-test-id="asset-bandwidth-limit" className={styles.includedLimit}>{`${withUnit(
          limit
        )} included`}</strong>
        {usage > limit && (
          <small className={styles.overage} data-test-id="asset-bandwidth-overage">
            {' '}
            + {withUnit(usage - limit)} overage
          </small>
        )}
        <TextLink
          href="https://www.contentful.com/r/knowledgebase/fair-use/"
          target="_blank"
          className={styles.learnMoreLink}>
          Fair Use Policy
        </TextLink>
        <Note className={styles.note}>
          Please note that asset bandwidth is calculated daily with a 48 hour delay.
        </Note>
      </div>
    </Typography>
  );
};

AssetBandwidthSection.propTypes = {
  limit: PropTypes.number.isRequired,
  usage: PropTypes.number.isRequired,
  uom: PropTypes.string.isRequired,
};

export default AssetBandwidthSection;
