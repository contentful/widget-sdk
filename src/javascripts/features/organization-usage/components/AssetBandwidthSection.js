import React from 'react';
import { css } from 'emotion';
import { partialRight } from 'lodash';
import {
  TextLink,
  Note,
  Typography,
  Heading,
  Paragraph,
} from '@contentful/forma-36-react-components';
import * as tokens from '@contentful/forma-36-tokens';

import { track } from 'analytics/Analytics';
import { shortenStorageUnit } from 'utils/NumberUtils';
import { useUsageState } from '../hooks/usageContext';

const styles = {
  heading: css({
    color: '#536171',
    fontWeight: tokens.fontWeightNormal,
  }),
  usageNumber: css({
    color: '#253545',
    fontSize: tokens.fontSize3Xl,
    fontWeight: tokens.fontWeightMedium,
  }),
  limit: css({
    fontSize: tokens.fontSizeM,
  }),
  includedLimit: css({
    color: '#6A7889',
  }),
  overage: css({
    color: tokens.colorOrangeMid,
  }),
  learnMoreLink: css({
    marginLeft: tokens.spacingS,
  }),
  note: css({
    width: '40%',
    marginTop: tokens.spacingS,
    color: '#27457A',
  }),
};

export const AssetBandwidthSection = () => {
  const { assetBandwidthData } = useUsageState();

  const { limit, usage, uom } = assetBandwidthData;
  const withUnit = partialRight(shortenStorageUnit, uom);
  const handleClick = () => {
    track('usage:fair_use_policy_clicked');
  };
  return (
    <Typography>
      <Heading element="h2" className={styles.heading}>
        Total asset bandwidth
      </Heading>
      <Paragraph data-test-id="asset-bandwidth-usage" className={styles.usageNumber}>
        {withUnit(usage)}
      </Paragraph>
      <div className={styles.limit}>
        <strong data-test-id="asset-bandwidth-limit" className={styles.includedLimit}>{`${withUnit(
          limit
        )} included`}</strong>
        {usage > limit && (
          <small className={styles.overage} data-test-id="asset-bandwidth-overage">
            {' '}
            + {withUnit(usage - limit)} overage
          </small>
        )}
        <TextLink
          href="https://www.contentful.com/r/knowledgebase/fair-use/"
          target="_blank"
          onClick={handleClick}
          data-test-id="fair_use_policy_link"
          rel="noopener noreferrer"
          className={styles.learnMoreLink}>
          Fair Use Policy
        </TextLink>
        <Note className={styles.note}>
          Please note that asset bandwidth is calculated daily with a 48 hour delay.
        </Note>
      </div>
    </Typography>
  );
};
