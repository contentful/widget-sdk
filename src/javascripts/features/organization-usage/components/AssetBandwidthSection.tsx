import React from 'react';
import { css } from 'emotion';
import { partialRight } from 'lodash';
import {
  TextLink,
  Note,
  Typography,
  Heading,
  Paragraph,
  SkeletonContainer,
  SkeletonDisplayText,
  SkeletonBodyText,
} from '@contentful/forma-36-react-components';
import tokens from '@contentful/forma-36-tokens';

import { track } from 'analytics/Analytics';
import { shortenStorageUnit } from 'utils/NumberUtils';
import { useUsageState } from '../hooks/usageContext';

const styles = {
  heading: css({
    color: '#536171',
    fontWeight: tokens.fontWeightNormal,
  }),
  usageNumber: css({
    color: '#253545',
    fontSize: tokens.fontSize3Xl,
    fontWeight: tokens.fontWeightMedium,
  }),
  limit: css({
    fontSize: tokens.fontSizeM,
  }),
  includedLimit: css({
    color: '#6A7889',
  }),
  overage: css({
    color: tokens.colorOrangeMid,
  }),
  learnMoreLink: css({
    marginLeft: tokens.spacingS,
  }),
  note: css({
    width: '40%',
    marginTop: tokens.spacingS,
    color: '#27457A',
  }),
};

export const AssetBandwidthSection = () => {
  const { assetBandwidthData, isLoading } = useUsageState();

  const { limit, usage, uom } = assetBandwidthData ?? {};

  // Show loading state once at the initial data fetch
  const isPageLoading = isLoading && !assetBandwidthData;

  const withUnit = partialRight(shortenStorageUnit, uom);
  const handleClick = () => {
    track('usage:fair_use_policy_clicked', { source: 'Asset Bandwidth' });
  };

  const UsageAndLimit = () => (
    <>
      <Paragraph data-test-id="asset-bandwidth-usage" className={styles.usageNumber}>
        {
          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
          // @ts-ignore
          withUnit(usage)
        }
      </Paragraph>
      <div className={styles.limit}>
        <strong data-test-id="asset-bandwidth-limit" className={styles.includedLimit}>{
          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
          // @ts-ignore
          `${withUnit(limit)} included`
        }</strong>
        {usage > limit && (
          <small className={styles.overage} data-test-id="asset-bandwidth-overage">
            {' '}
            +{' '}
            {
              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
              // @ts-ignore
              withUnit(usage - limit)
            }{' '}
            overage
          </small>
        )}
        <TextLink
          href="https://www.contentful.com/r/knowledgebase/fair-use/"
          target="_blank"
          onClick={handleClick}
          data-test-id="fair_use_policy_link"
          rel="noopener noreferrer"
          className={styles.learnMoreLink}>
          Fair Use Policy
        </TextLink>
      </div>
    </>
  );

  return (
    <Typography>
      <Heading element="h2" className={styles.heading}>
        Total asset bandwidth
      </Heading>
      {isPageLoading && (
        <SkeletonContainer svgWidth="40%" svgHeight="80px">
          <SkeletonDisplayText numberOfLines={1} />
          <SkeletonBodyText numberOfLines={1} offsetTop={50} />
        </SkeletonContainer>
      )}
      {!isPageLoading && <UsageAndLimit />}
      <Note className={styles.note}>
        Please note that asset bandwidth is calculated daily with a 48 hour delay.
      </Note>
    </Typography>
  );
};
