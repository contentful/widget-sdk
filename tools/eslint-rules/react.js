module.exports = {
  extends: ['plugin:react/recommended', 'prettier/react'],
  plugins: ['react-hooks'],
  settings: {
    react: {
      version: '16.10.2',
    },
  },
  rules: {
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'error',
    'react/display-name': 'off',
    'react/prop-types': ['error', { ignore: ['children'] }],
    'react/no-unused-prop-types': 'error',
    'react/jsx-no-target-blank': ['error', { enforceDynamicLinks: 'always' }],
  },
};
