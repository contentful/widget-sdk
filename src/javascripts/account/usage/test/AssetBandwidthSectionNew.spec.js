import '@testing-library/dom';
import React from 'react';
import { render } from '@testing-library/react';
import AssetBandwidthSection from '../committed/AssetBandwidthSectionNew';

describe('AssetBandwidthSectionNew', () => {
  const defaultProps = {
    limit: 750,
    usage: 200,
    uom: 'GB'
  };

  const renderComp = props => {
    return render(<AssetBandwidthSection {...props} />);
  };

  it('should render', () => {
    const { container } = render();
    expect(container).toMatchSnapshot();
  });

  it('should render correct data', () => {
    const { getByTestId } = renderComp(defaultProps);
    const bandwidthUsage = getByTestId('asset-bandwidth-usage');
    const bandwidthLimit = getByTestId('asset-bandwidth-limit');
    expect(bandwidthUsage.textContent).toBe('200 GB');
    expect(bandwidthLimit.textContent).toBe('750 GB included');
  });

  it('should render bandwidth data not available', () => {
    const undefinedProps = {
      limit: undefined,
      usage: undefined,
      uom: undefined
    };

    const { getByTestId } = renderComp(undefinedProps);
    const notAvilable = getByTestId('asset-bandwidth-not-available');
    expect(notAvilable.textContent).toBe('Not available');
  });

  it('should render overage with correct data', () => {
    const overageProps = {
      limit: 200,
      usage: 750,
      uom: 'GB'
    };
    const { getByTestId } = renderComp(overageProps);
    const overage = getByTestId('asset-bandwidth-overage');
    expect(overage.textContent).toBe(' + 550 GB overage');
  });
});
