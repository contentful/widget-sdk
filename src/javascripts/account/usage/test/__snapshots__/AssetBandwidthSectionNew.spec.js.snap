// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetBandwidthSectionNew should render 1`] = `<div />`;
